frontend/src/lib/utils/textSize.ts

```diff
@@ -1,12 +1,38 @@
-export function getTextWidthWithFont(text: string, font: string): number {
-    // re-use canvas object for better performance
+type CharWidths = Map<string, number>;
+
+const text = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789-_.!~*'()[]{}/?:@#$%^&*+=`|\\\"<>";
```

Comment:
Consider renaming to `textChars`, `validChars`, etc, to deviate from the `text` argument of `getTextWidthWithFont`-function below.